app/services/hyrax/analytics/ga4_decorator.rb

```diff
@@ -0,0 +1,315 @@
+# frozen_string_literal: true
+
+begin
```

Comment:
This decorator is HUGE. How can we clean this up? What can be contributed back to Hyrax?